### PR TITLE
Feature/field get set bindings

### PR DESCRIPTION
**Pull request by gfodor**
This PR adds a number of performance improvements by increasing the surface area of fast paths for code generated bindings for CLR types. These include:

- Field getter + setters
- Default instantiation and array instantiation
- Memberwise cloning

### CODE REVIEW

File: ILRuntime/Runtime/Intepreter/ILIntepreter.cs [hunk 1]
```diff
@@ -1924,7 +1926,7 @@ public object Run(ILMethod method, object instance, object[] p)
                                             int idx = (int)ip->TokenLong;
                                             var f = t.GetField(idx);
```
Comment (liiir1985) [inline ILRuntime/Runtime/Intepreter/ILIntepreter.cs]: this f here is not used anymore

File: ILRuntime/Runtime/Intepreter/ILIntepreter.cs [hunk 2]
```diff
@@ -1948,7 +1950,7 @@ public object Run(ILMethod method, object instance, object[] p)
                                             CLRType t = type as CLRType;
                                             int idx = (int)ip->TokenLong;
                                             var f = t.GetField(idx);
```
Comment (liiir1985) [inline ILRuntime/Runtime/Intepreter/ILIntepreter.cs]: This f here is not used anymore

File: ILRuntime/Runtime/Intepreter/ILIntepreter.cs (hunk 1 above)
Comment (gfodor) [inline ILRuntime/Runtime/Intepreter/ILIntepreter.cs]: I believe it is used a few lines below for `FieldType`. It'd be nice to be able to remove it though if possible.

File: ILRuntime/Runtime/Intepreter/ILIntepreter.cs (hunk 2 above)
Comment (gfodor) [inline ILRuntime/Runtime/Intepreter/ILIntepreter.cs]: Used to determine below if it is a boxed value (by checking `FieldType == typeof(object)`)

**Comment by liiir1985** [comment]
Hmm, it seems it has now conflicts

**Pull request merged**